src/Catalog/Registration/RegistrationMakerCatalogItem.cs

```diff
@@ -155,6 +157,43 @@ private Uri GetPackageContentAddress()
             return _packageContentAddress;
         }
 
+        private string GetLicenseUrl()
+        {
+            INode subject = _catalogItem.CreateUriNode(_catalogUri);
+
+            string id = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.Id)).FirstOrDefault().Object.ToString();
+            string version = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.Version)).FirstOrDefault().Object.ToString();
+            Triple licenseExpression = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.LicenseExpression)).FirstOrDefault();
+            Triple licenseFile = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.LicenseFile)).FirstOrDefault();
+            Triple licenseUrl = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.LicenseUrl)).FirstOrDefault();
+
+            if (_galleryBaseAddress != null)
+            {
+                if (licenseExpression != null && !string.IsNullOrWhiteSpace(licenseExpression.Object.ToString()))
+                {
+                    return GetOverWriteLicenseUrl(id, version);
+                }
+                if (licenseFile != null && !string.IsNullOrWhiteSpace(licenseFile.Object.ToString()))
+                {
+                    return GetOverWriteLicenseUrl(id, version);
+                }
+            }
+
+            if (licenseUrl != null)
+            {
+                return licenseUrl.Object.ToString();
+            }
+
+            return string.Empty;
+        }
+
+        private string GetOverWriteLicenseUrl(string id, string version)
```

Comment:
I don't understand this method name. What is "OverWrite"?

Reply:
Change to "GetGalleryLicenseUrl" now.